src/main/java/com/github/cameltooling/lsp/internal/kubernetes/KubernetesConfigManager.java

```diff
@@ -36,8 +37,15 @@ public void setClient(KubernetesClient client) {
 	}
 	
 	public KubernetesClient getClient() {
-		if(client == null) {
+		if(client == null ) {
 			client = new KubernetesClientBuilder().build();
+		} else {
+			try {
+				//Let's see if the connection is still valid
+				client.namespaces().list();
```

Comment:
thoughts: that's ugly as hell and will throw an exception if missing rights too. But I have not found a method to check if the connection is still valid.